src/tasks/generate-verifiers.ts

```diff
@@ -104,6 +69,81 @@ export const generateVerifiers: ActionType<GenerateVerifiersTaskConfig> = async
   }
 };
 
+async function getAndCountArtifacts(
+  env: HardhatRuntimeEnvironment,
+  allFullyQualifiedNames: string[],
+): Promise<
+  [
+    { [key: string]: number },
+    {
+      name: string;
+      circuitArtifact: CircuitArtifact;
+    }[],
+  ]
+> {
+  const templateNamesCount: { [key: string]: number } = {};
+  const circuitArtifactsInfo = await Promise.all(
+    allFullyQualifiedNames.map(async (name: string) => {
+      const circuitArtifact: CircuitArtifact = await env.zkit.circuitArtifacts.readCircuitArtifact(name);
+
+      // Count the number of uses of templates to determine whether a verifier file
+      // will need a suffix in case of non-uniqueness
+      templateNamesCount[circuitArtifact.circuitTemplateName] =
+        (templateNamesCount[circuitArtifact.circuitTemplateName] || 0) + 1;
+
+      return { name, circuitArtifact };
+    }),
+  );
+
+  return [templateNamesCount, circuitArtifactsInfo];
+}
+
+async function generateVerifierForProvingSystem(
+  taskArgs: GenerateVerifiersTaskConfig,
+  env: HardhatRuntimeEnvironment,
+  circuitArtifactInfo: {
+    name: string;
+    circuitArtifact: CircuitArtifact;
+  },
+  provingSystems: ProvingSystemType[],
```

Comment:
We can pass the total number of proving systems, as this parameter is only used for this purpose. In the current form slightly confuse the two parameters `provingSystem` and `provingSystems`.